core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -856,6 +856,9 @@ private[spark] class SparkSubmit extends Logging {
     if (args.verbose && isSqlShell(childMainClass)) {
       childArgs ++= Seq("--verbose")
     }
+
+    sparkConf.set("spark.app.submitTime", System.currentTimeMillis().toString)
```

Comment:
do we also report the start time using this way (set a config) ?

Reply:
This seems specifically only for `Environment`'s `Spark Properties` table.

https://github.com/apache/spark/blob/122a59986f6ab358452a293256ad45be0fe96a1c/core/src/main/scala/org/apache/spark/ui/env/EnvironmentPage.scala#L77-L79

Reply:
`spark.app.startTime` is also setting this, but `startTime` is propagated to more classes differently because it's used in non-UI usage.
```
_conf.set("spark.app.startTime", startTime.toString)
```

The following is another usage pattern of `startTime`.
```
Some(SparkUI.create(Some(this), _statusStore, _conf, _env.securityManager, appName, "", startTime))
```

```
listenerBus.post(SparkListenerApplicationStart(appName, Some(applicationId),
      startTime, sparkUser, applicationAttemptId, schedulerBackend.getDriverLogUrls,
      schedulerBackend.getDriverAttributes))
```

Reply:
Yes. It is set in [SparkContext.scala#393](https://github.com/apache/spark/blob/4edc5d3263c6be782a48a667e252a27ed622b363/core/src/main/scala/org/apache/spark/SparkContext.scala#L393)